src/crafting.h

```diff
@@ -104,6 +99,51 @@ struct recipe {
 
 };
 
+class recipe_dictionary {
```

Comment:
That's a nice container class. It could (in the future, probably not in this PR) also house the `find_recipe`, `load_recipe` and similar functions from below.